### PR TITLE
Rating implementation

### PR DESCRIPTION
**Pull request by pboettch**
A fixed field called rating is implemented, accepting values from 1 to 10, or None if not set.

### CODE REVIEW

**Comment by jeandet** [comment]
@pboettch it looks ok to me is this ready for merge?

**Comment by pboettch** [comment]
No, let me finish the work on tscat_gui with this change, to see whether it fits all the needs. In the best case, no more changes necessary here.

File: tests/test_event.py [hunk 1]
```diff
@@ -102,3 +102,30 @@ def test_constructor_with_dynamic_attribute_manual_access(self):
         self.assertEqual(e.field_str, "string-test")
         self.assertEqual(e.field_bool, True)
         self.assertEqual(e.field_dt, dt_val)
+
+    def test_event_rating(self):
```
Comment (jeandet) [inline tests/test_event.py:106]: Should be split in two data driven tests.

File: tscat/orm_sqlalchemy/orm.py [hunk 2]
```diff
@@ -34,18 +34,20 @@ class Event(Base):
...
         self.start = start
         self.stop = stop
         self.author = author
         self.uuid = uuid
         self.tags = tags
         self.products = products
+        self.rating = rating
         self.attributes = attributes
 
     def __repr__(self):  # pragma: no cover
```
Comment (jeandet) [inline tscat/orm_sqlalchemy/orm.py:53]: Why no cover and why not all attributes?

File: tscat/base.py [hunk 3]
```diff
@@ -193,10 +196,16 @@ def __setattr__(self, key, value):
             if value < self.start:
                 raise ValueError("stop date has to be after start date")
         elif key in ['tags', 'products']:
-            if any(type(v) != str for v in value):
+            if any(not isinstance(v, str) for v in value):
```
Comment (jeandet) [inline tscat/base.py:199]: Looking at this, I wonder if using properties isn't better, each property would holds its own logic.

Comment (pboettch) [reply tscat/base.py:199]: One requirement for tscat was to be able to do `event.aribtrary_field = "value"`  for dynamic attributes and fixed-attributes. At the time of writing the `__setattr__`-function it was the only way I found.

I might have to look at the `@property`-decorator again, but AFAICR the only way to set a property is by using a setter-function? Or does this allow also the use of the  assignment operator.

**Pull request merged**